pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -159,80 +163,94 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 		}
 	}
 
-	klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	if len(addedPermissionClaims) != 0 {
+		klog.V(2).Infof("Adding permission claims %v for APIBinding %s|%s", addedPermissionClaims, lc, apiBinding.Name)
+	}
 
 	var allPatchErrors []error
-	for gvr, claims := range addedPermissionClaims {
+	// For all the things that are in the accepted claims we need to re-sync
+	for gvr, claims := range syncClaims {
 		for _, claim := range claims {
-			// TODO: use indexer instead of lister.
-			objs, err := claim.lister.Lister().List(labels.Everything())
-			if err != nil {
-				return err
+			//TODO: use indexer instead of lister.
+			if !claim.informer.Informer().HasSynced() {
+				return fmt.Errorf("unable sync cache")
 			}
 			var errs []error
+			selector := labels.NewSelector()
+			req, err := labels.NewRequirement(claim.key, selection.NotEquals, []string{claim.label})
+			if err != nil {
+				errs = append(errs, err)
+				klog.V(4).Infof("uanble to get requirement: %v ", err)
+			}
+			selector.Add(*req)
+			objs, err := claim.informer.Lister().List(selector)
```

Comment:
🔴  we can't do this. It will possibly return many thousands of objects.

Reply:
use the index ByWorkspace and loop there. A magnitude (or more) faster.

Reply:
done updated